x-pack/plugins/security_solution/public/flyout/document_details/shared/constants/panel_keys.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IsolateHostPanelProps } from '../../isolate_host';
+import type { LeftPanelProps } from '../../left';
+import type { PreviewPanelProps } from '../../preview';
+import type { RightPanelProps } from '../../right';
+
+export const DocumentDetailsRightPanelKey: RightPanelProps['key'] = 'document-details-right';
+export const DocumentDetailsLeftPanelKey: LeftPanelProps['key'] = 'document-details-left';
+export const DocumentDetailsPreviewPanelKey: PreviewPanelProps['key'] = 'document-details-preview';
+export const DocumentDetailsIsolateHostPanelKey: IsolateHostPanelProps['key'] =
+  'document-details-isolate-host';
```

Comment:
NIT: I see these `key` constants are hardcoded twice, once in this file and also in the type definitions. We could avoid that duplication by using these constant values to create the types (e.g. `key: typeof DocumentDetailsRightPanelKey;`), instead of typing the constants, which is a bit redundant. So this file would become the single source of truth.

Reply:
@semd is this what you had in mind? https://github.com/elastic/kibana/pull/181497/commits/f77a9f2a6e77b05bfa0c7eaa54ffd3313e92946f

Reply:
Exactly! 👍 thanks